content/docs/concepts/oakestra-extensions/addons.md

```diff
@@ -0,0 +1,87 @@
+
+Addons fall into two primary categories in Oakestra:
+- **Plugins**: Components that replace or augment core functionalities.
+- **Extensions**: Components that introduce new capabilities outside of the core functionalities.
+
+This modular design enables developers to activate or deactivate features as needed, maintaining a lightweight, adaptable system suitable for diverse operational contexts.
+
+## Plugins vs Extensions
+
+Although *plugins* and *extensions* are often used interchangeably to describe system extensibility, in Oakestra, they serve distinct roles:
+- A **plugin** is a component that integrates directly with a core functionality, often replacing or augmenting an essential system module.
+- An **extension**, on the other hand, enhances the system capabilities by adding optional features without altering any core component.
+
+To simplify, Oakestra refers to both plugins and extensions collectively as **addons**, a term that represents any additional component that can enhance or modify Oakestras capabilities.
+
+### Plugin Example
+
+In Oakestra, plugins provide a way to swap out components, enhancing the systems adaptability. For example, the default scheduler could be replaced with an alternative that uses a more advanced algorithm. Consider a scheduler that accounts for environmental variables, such as energy consumption or green energy sources, when making scheduling decisions. This plugin can be enabled by users who need these advanced features, offering them a customized scheduling tool while other users retain the default, core scheduler. This approach keeps the core system minimal while catering to varied user requirements.
+
+### Extensions Example
+
+Extensions in Oakestra can add substantial new capabilities to the system. One example could be the introduction of federated learning on edge computing devices. Federated learning enables decentralized training across multiple devices, enhancing privacy by not requiring raw data transfer. Implementing this as an extension involves adding new components to the Orchestrator control plane for the coordination and aggregation of distributed training processes. Users can activate or deactivate this extension based on their needs, making it a flexible addition that adapts Oakestra to specialized use cases.
+
+## Addon System Design
+
+Addons in Oakestra are modular components that extend the platforms core functionality while keeping the foundational architecture lightweight. This design ensures that Oakestra remains efficient, avoiding unnecessary bloat in edge computing environments.
+
+Addons system is composed of two subsystems:
+- **Addons Marketplace**: Consists of Marketplace Manager and MongoDB client.
+- **Addons Engine**: Consists of Addons Manager, Addons Monitor, and MongoDB Client.
```

Comment:
Maybe remove these lines here and bring the short description to the section itself?